mslice/models/slice/mantid_slice_algorithm.py

```diff
@@ -32,14 +31,35 @@ def compute_slice(self, selected_workspace, x_axis, y_axis, smoothing, norm_to_o
         # perform number of events normalization then mask cells where no data was found
         with np.errstate(invalid='ignore'):
             plot_data = thisslice.getSignalArray() / thisslice.getNumEventsArray()
-        plot_data = np.ma.masked_where(np.isnan(plot_data), plot_data)
         # rot90 switches the x and y axis to to plot what user expected.
         plot_data = np.rot90(plot_data)
         self._workspace_provider.delete_workspace(thisslice)
         boundaries = [x_axis.start, x_axis.end, y_axis.start, y_axis.end]
         if norm_to_one:
             plot_data = self._norm_to_one(plot_data)
-        return plot_data, boundaries
+        plot = [None, None, None]
+        plot[0] = plot_data
```

Comment:
For larger datasets it might be a problem that we are calculating all 3 up front - this would be a waste if they are never used. Can we calculate the `chi` values upon first use and then cache?